hazelcast/src/test/java/com/hazelcast/cache/CachingProviderTest.java

```diff
@@ -132,6 +132,14 @@ public void whenConfigLocationAsUri_thenThatInstanceIsUsed() throws URISyntaxExc
         assertCacheManagerInstance(cacheManager, instance3);
     }
 
+    @Test
+    public void whenConfigLocationAsUriViaProperty_thenThatInstanceIsUsed() throws URISyntaxException {
+        System.setProperty("PROPERTY_PLACEHOLDER", "classpath:" + CONFIG_CLASSPATH_LOCATION);
```

Comment:
The test runner ensures system properties are [restored afterwards](https://github.com/vbekiaris/hazelcast/blob/77ee1969666d385b1bcbc6a6294911f398b7e9fd/hazelcast/src/test/java/com/hazelcast/test/HazelcastSerialClassRunner.java#L56).